hudi-hadoop-mr/src/main/java/org/apache/hudi/hadoop/realtime/HoodieParquetRealtimeInputFormat.java

```diff
@@ -69,6 +69,8 @@ public RecordReader<NullWritable, ArrayWritable> getRecordReader(final InputSpli
     HoodieTableMetaClient metaClient = HoodieTableMetaClient.builder().setConf(jobConf).setBasePath(realtimeSplit.getBasePath()).build();
     HoodieTableConfig tableConfig = metaClient.getTableConfig();
     addProjectionToJobConf(realtimeSplit, jobConf, metaClient.getTableConfig().getPreCombineField());
+    // add partition fields to hive job conf
+    HoodieRealtimeInputFormatUtils.addProjectionField(jobConf, metaClient.getTableConfig().getPartitionFields());
```

Comment:
Adding partition fields to READ_COLUMN_IDS_CONF_STR affects the performance, 
the problem may be solved by hive via https://github.com/apache/hive/pull/3742.